pyctcdecode/decoder.py

```diff
@@ -609,6 +665,65 @@ def _decode_logits(
         ]
         return output_beams
 
+    def get_starting_state(self) -> Tuple[List[Beam], LMScoreCache, Dict[str, float]]:
+        """Get the starting beams and initial caches."""
+        start_beam = [EMPTY_START_BEAM]
+        language_model = self._language_model
+        if language_model is None:
+            cached_lm_scores: LMScoreCache = {}
+        else:
+            start_state = language_model.get_start_state()
+            cached_lm_scores = {("", False): (0.0, 0.0, start_state)}
+        cached_p_lm_scores: Dict[str, float] = {}
+        return start_beam, cached_lm_scores, cached_p_lm_scores
+
+    def partial_decode_beams(
+        self,
+        logits: NDArray[NpFloat],
+        cached_lm_scores: LMScoreCache,
+        cached_p_lm_scores: Dict[str, float],
+        beams: List[Beam],
+        beam_width: int = DEFAULT_BEAM_WIDTH,
+        beam_prune_logp: float = DEFAULT_PRUNE_LOGP,
+        token_min_logp: float = DEFAULT_MIN_TOKEN_LOGP,
+        prune_history: bool = DEFAULT_PRUNE_BEAMS,
+        hotword_scorer: Optional[HotwordScorer] = None,
```

Comment:
here I allow the user to directly input a hotword scorer object. in the case of partial decoding, a typical use case would be to repeatedly use the same hotwords between calls to this function for a single audio file, so we don't want to be building an identical hotword scorer over and over

Reply:
This is neat, too, as one could also use different hotword scorers over the course of an audio file if they have some reasonable prior about word distribution